docs/04-operation-guides/troubleshooting/eventing/evnt-05-fix-pending-messages.md

```diff
@@ -8,139 +8,138 @@ You publish events, but some of them are not received by the subscriber and stay
 As a result of this, messages on some consumers can stop being acknowledged and start piling up in the stream.
 
 ## Remedy
 
-There are two ways of how to fix the "broken" consumers with pending messages. You will need either to trigger a leader
-reelection either on the consumers
-with pending messages or on the stream level.
+To fix the "broken" consumers with pending messages, trigger a leader reelection. You can do this either on the consumer that have pending messages, or if that fails, on the stream level.
 
-### Trigger consumer leader election
+You need the latest version of NATS cli installed on your machine.
 
-First, you need to find out which consumer(s) have pending messages. For that, you need the latest version of NATS cli
-installed on your machine.
-You can find the broken consumer in two ways: by using Grafana dashboard or directly by using the NATS cli command.
+### Consumer leader reelection
 
-#### Find the broken consumers using Grafana dashboard
+First, find out which consumer(s) have pending messages. You can find the broken consumer either with the NATS cli command, or with a Grafana dashboard.
 
-1. [Access and Expose Grafana](../../security/sec-06-access-expose-grafana.md)
-2. Find the NATS JetStream Dashboard and check the pending messages
-   ![Pending consumer](../../assets/grafana_pending_consumer.png)
-3. Find the consumer with pending messages and encode it as an `md5` hash:
-
-```bash
-echo -n "tunas-testing/test-noapp3/kyma.noapp.order.created.v1" | md5
-```
+#### Option 1: Find the broken consumers with NATS cli
 
-this shell command results in `ebcabfe5c902612f0ba3ebde7653f30b`.
-
-4. Port forward to a NATS replica:
+1. Port forward to a NATS replica:
 
    ```bash
    kubectl port-forward -n kyma-system eventing-nats-0 4222  
 
-5. Find the consumer's leader:
+2. Run this shell script:
 
-```bash
-nats consumer info sap ebcabfe5c902612f0ba3ebde7653f30b
-```
+   ```bash
+   for consumer in $(nats consumer list -n sap) # sap is the stream name
+   do
+     nats consumer info sap $consumer -j | jq -c '{name: .name, pending: .num_pending, leader: .cluster.leader}'
+   done
+   ```
 
-```bash
-Information for Consumer sap > ebcabfe5c902612f0ba3ebde7653f30b created 2022-10-24T15:49:43+02:00
+   You get an output like the following:
 
-Configuration:
+   ```bash
+   {"name":"ebcabfe5c902612f0ba3ebde7653f30b","pending":25,"leader":"eventing-nats-1"}
+   {"name":"c74c20756af53b592f87edebff67bdf8","pending":0,"leader":"eventing-nats-0"}
+   ```
 
-                Name: ebcabfe5c902612f0ba3ebde7653f30b
-         Description: tunas-testing/test-noapp3/kyma.noapp.order.created.v1
-          ...
+3. Check the output to see which consumer has pending messages and which replica is the leader.
+   In this example, the consumer `ebcabfe5c902612f0ba3ebde7653f30b` has 25 pending messages and has the leader.
+   The other one has no pending message and is successfully processing events.
```

Comment:
Maybe I get it wrong but `c74c20756af53b592f87edebff67bdf8` also seems to have the leader, doesn't it 🤔?  Or we're not interested in that piece of information because what interests us is only the broken customer with piled messages?

Reply:
I think we're only interested in the leader of the broken consumer, because we don't have to reelect the leader for the working consumer.